### PR TITLE
[Core] assure uniqueness for new node guid

### PR DESCRIPTION
**Pull request by rodolforg**

### CODE REVIEW

**Comment by rodolforg** [comment]
Not probable, but not impossible to happen…

Maybe backport to 1.4.1?

File: synfig-core/src/synfig/node.cpp [hunk 1]
```diff
@@ -202,8 +204,11 @@ Node::get_guid()const
 	std::lock_guard<std::mutex> lock(guid_mutex_);
 	if(!guid_)
 	{
-		guid_.make_unique();
-		global_node_map().add(guid_, const_cast<Node*>(this));
+		bool added = false;
```
Comment (ice0) [inline synfig-core/src/synfig/node.cpp:207]: Maybe shorten this code?
```
do {
	guid_.make_unique();
} while (!global_node_map().add(guid_, const_cast<Node*>(this)));
```

Comment (rodolforg) [reply synfig-core/src/synfig/node.cpp:207]: Yeah, I thought it this way, but maybe not that readable.
Compiler would throw that variable away anyway, right?

Comment (ice0) [reply synfig-core/src/synfig/node.cpp:207]: > Compiler would throw that variable away anyway, right?
Yes.

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**